File: Tweet-Lookup/get_tweets_with_bearer_token.rb
# This script uses your bearer token to authenticate and retrieve the specified Tweet objects (by ID)

require 'json'
require 'typhoeus'

# The code below sets the bearer token from your environment variables
# To set environment variables on Mac OS X, run the export command below from the terminal:
# export BEARER_TOKEN='YOUR-TOKEN'
bearer_token = ENV["BEARER_TOKEN"]

tweet_lookup_url = "https://api.twitter.com/2/tweets"

# Specify the Tweet IDs that you want to lookup below (to 100 per request)
tweet_ids = "1261326399320715264,1278347468690915330"

# Add or remove optional parameters values from the params object below. Full list of parameters and their values can be found in the docs:
# https://developer.twitter.com/en/docs/twitter-api/tweets/lookup/api-reference
params = {
	"ids": tweet_ids,
  # "expansions": "author_id,referenced_tweets.id",
  "tweet.fields": "attachments,author_id,conversation_id,created_at,entities,geo,id,in_reply_to_user_id,lang",
  # "user.fields": "name"
  # "media.fields": "url", 
  # "place.fields": "country_code",
  # "poll.fields": "options"
}

def tweet_lookup(url, bearer_token, params)
  options = {
    method: 'get',
    headers: {
      "User-Agent": "v2TweetLookupRuby",
      "Authorization": "Bearer #{bearer_token}"
    },
    params: params
  }

  request = Typhoeus::Request.new(url, options)
  response = request.run

  return response
end

response = tweet_lookup(tweet_lookup_url, bearer_token, params)
puts response.code, JSON.pretty_generate(JSON.parse(response.body))

File: Tweet-Lookup/get_tweets_with_bearer_token.go
package Tweet_Lookup

// Get Tweets by User ID. Using Bearer Token
// https://developer.twitter.com/en/docs/twitter-api/tweets/lookup/introduction

import (
	"encoding/json"
	"fmt"
	"log"
	"net/http"
	"os"
)

type UserTweets struct {
	Data     []Data   `json:"data"`
	Includes Includes `json:"includes"`
}
type Data struct {
	Lang     string `json:"lang"`
	AuthorID int64  `json:"author_id"`
	ID       int64  `json:"id"`
	Text     string `json:"text"`
}
type Users struct {
	ID       int64  `json:"id"`
	Name     string `json:"name"`
	Username string `json:"username"`
}
type Includes struct {
	Users []Users `json:"users"`
}

func TweetLookup() {
	if os.Getenv("Bearer") == "" {
		log.Fatalln("set a bereaer token")
		return
	}

	// Sets the bearer token from your environment variables
	// To set environment variables on macOS or Linux, run the export command below from the terminal:
	// export BEARER_TOKEN='YOUR-TOKEN'
	bearerToken := os.Getenv("BEARER_TOKEN")
	const url = "https://api.twitter.com/2/tweets"

	client := &http.Client{}
	request, createReqErr := http.NewRequest("GET", url, nil)
	if createReqErr != nil {
		log.Println(createReqErr)
		return
	}

	// Add Parameters
	q := request.URL.Query()
	q.Add("ids", "1228393702244134912,1227640996038684673,1199786642791452673")
	q.Add("tweet.fields", "lang,author_id")
	q.Add("expansions", "author_id")

	request.URL.RawQuery = q.Encode()

	// Add Bearer Token to HTTP Header
	request.Header.Set("Authorization", "Bearer "+bearerToken)

	// Send Request
	response, responseErr := client.Do(request)
	if responseErr != nil {
		log.Println(responseErr)
	}

	defer response.Body.Close()

	var userTweets UserTweets
	json.NewDecoder(response.Body).Decode(&userTweets)

	fmt.Println(userTweets)
}
